### DISCUSSION TITLE
issue: default values persist after unregister

### DISCUSSION THREAD

**Discussion by dfernandezfb** [react-hook-form/react-hook-form#9975] [Q&A]
### Version Number

7.42.1

### Codesandbox/Expo snack

https://codesandbox.io/s/trusting-field-o8xlcp?file=/src/App.js

### Steps to reproduce

1. Go to [sandbox](https://codesandbox.io/s/trusting-field-o8xlcp?file=/src/App.js)
2. Click on "Reset" button
3. Click on "Option 1" radio button
4. Click on "Option 2" radio button
5. Input still has its value set by `reset`

### Expected behaviour

The input's value should be cleared after unregister

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@dfernandezfb 

You are using `reset` method with option `keepDefaultValues: false` therefore default values will be set to your new values including the input `suboption`.

Flag `shouldUnregister: true` has nothing to do with default values. If the target input is unmount then mount again, the input element will be registered with new default value that you've set on `reset` method.
